### PR TITLE
add hotspots picker for sorting hotspots

### PR DESCRIPTION
**Pull request by allenan**

### CODE REVIEW

File: src/store/hotspots/hotspotsSlice.ts [hunk 1]
```diff
@@ -1,7 +1,40 @@
...
+  Near = 'near',
+  Earn = 'earn',
+  Offline = 'offline',
+}
+
+type Rewards = Record<string, Sum>
+
+type SorterContext = { rewards?: Rewards; location?: LocationCoords }
+type HotspotSorter = (hotspots: Hotspot[], context?: SorterContext) => Hotspot[]
+const hotspotSorters: Record<HotspotSort, HotspotSorter> = {
```
Comment (tyler-whitman) [inline src/store/hotspots/hotspotsSlice.ts:19]: Nice, I like how you set this up

**Review by tyler-whitman — APPROVED** [review]
Working well!

File: src/utils/location.ts [hunk 2]
```diff
@@ -9,3 +11,29 @@ export const getCurrentPosition = async (
   const pos = await Location.getCurrentPositionAsync({ accuracy })
   return pos.coords
 }
+
+function toRad(x: number): number {
+  return (x * Math.PI) / 180
+}
+
+export const distance = (coords1: LocationCoords, coords2: LocationCoords) => {
```
Comment (matthewcarlreetz) [inline src/utils/location.ts:19]: is this haversine distance?

**Review by matthewcarlreetz — APPROVED** [review]
Looks good!

File: src/utils/location.ts (hunk 2 above)
Comment (allenan) [reply src/utils/location.ts:19]: yeah, seems to work for my hotspots

File: src/store/hotspots/hotspotsSlice.ts (hunk 1 above)
Comment (allenan) [reply src/store/hotspots/hotspotsSlice.ts:19]: thanks, it was a little awkward getting some of the additional context in there, like the rewards and location, but seems to work ok

**Pull request merged**